packages/vite/src/node/plugins/css.ts

```diff
@@ -1399,12 +1432,45 @@ export interface PreprocessCSSResult {
 /**
  * @experimental
  */
+export function createCSSPreprocessor(): {
```

Comment:
Been thinking about this one, initially I though WeakMap of the `ResolvedConfig` to the controller could work here, but I guess that won't work for manually called `resolveConfig` usage. It would be nice if the preprocessor could share the controller created by the CSS plugin, but I guess also it makes it complicated if you call it at the wrong time before the controller is initialized.

Checking how `vite-plugin-svelte` uses it, I think this pattern will be a bit tricky to use since it creates a standalone Svelte preprocessor (an object with functions as values), and we would have to expose some sort `close()` function too, which isn't common in Svelte preprocessors, and there's also no way to know if something ends, e.g. Svelte VSCode could be calling that too.

I think my proposal is that:
1. Keep `preprocessCSS()`. If the ResolvedConfig has a related controller, use that. Otherwise fallback to running in the main thread.
2. Also expose `createCSSPreprocessor()` for users who want more control. I don't have a need to use this for Svelte and Astro for now though if we implement no1.

Reply:
Awesome. Thanks!